exchanges/gateio/gateio_wrapper.go

```diff
@@ -833,116 +1646,299 @@ func (g *Gateio) GetOrderHistory(ctx context.Context, req *order.GetOrdersReques
+				!strings.EqualFold(formattedPair.String(), currency.BTC.String()) {
+				return klineData, errUnsupportedSettleValue
+			}
 
-	klineData.SortCandlesByTimestamp(false)
-	klineData.RemoveOutsideRange(start, end)
+			var candles []FuturesCandlestick
+			if a == asset.Futures {
+				candles, err = g.GetFuturesCandlesticks(ctx, formattedPair.Quote.Lower().String(), pair, dates.Ranges[b].Start.Time, dates.Ranges[b].End.Time, uint64(g.Features.Enabled.Kline.ResultLimit), interval)
+			} else {
+				candles, err = g.GetDeliveryFuturesCandlesticks(ctx, formattedPair.Quote.Lower().String(), formattedPair.Upper().String(), dates.Ranges[b].Start.Time, dates.Ranges[b].End.Time, uint64(g.Features.Enabled.Kline.ResultLimit), interval)
+			}
+			if err != nil {
+				return klineData, err
+			}
+			for x := range candles {
+				candlestickItems = append(candlestickItems, kline.Candle{
+					Time:   candles[x].Timestamp,
+					Open:   candles[x].OpenPrice,
+					High:   candles[x].HighestPrice,
+					Low:    candles[x].LowestPrice,
+					Close:  candles[x].ClosePrice,
+					Volume: candles[x].Volume,
+				})
+			}
+			return klineData, nil
+		case asset.Options:
+			candles, err := g.GetOptionFuturesCandlesticks(ctx, formattedPair.Upper().String(), uint64(g.Features.Enabled.Kline.ResultLimit), start, end, interval)
+			if err != nil {
+				return klineData, err
+			}
+			for x := range candles {
+				candlestickItems = append(candlestickItems, kline.Candle{
+					Time:   candles[x].Timestamp,
+					Open:   candles[x].OpenPrice,
+					High:   candles[x].HighestPrice,
+					Low:    candles[x].LowestPrice,
+					Close:  candles[x].ClosePrice,
+					Volume: candles[x].Volume,
+				})
+			}
+			return klineData, nil
+		default:
+			return klineData, fmt.Errorf("%s does not support %s", g.Name, a)
+		}
+	}
+	klineData.Candles = candlestickItems
+	if start.IsZero() || end.IsZero() {
+		klineData.SortCandlesByTimestamp(false)
+		klineData.RemoveOutsideRange(start, end)
+	}
 	return klineData, nil
 }
 
-// GetHistoricCandlesExtended returns candles between a time period for a set time interval
-func (g *Gateio) GetHistoricCandlesExtended(ctx context.Context, pair currency.Pair, a asset.Item, start, end time.Time, interval kline.Interval) (kline.Item, error) {
-	return g.GetHistoricCandles(ctx, pair, a, start, end, interval)
-}
-
 // GetAvailableTransferChains returns the available transfer blockchains for the specific
 // cryptocurrency
 func (g *Gateio) GetAvailableTransferChains(ctx context.Context, cryptocurrency currency.Code) ([]string, error) {
-	chains, err := g.GetCryptoDepositAddress(ctx, cryptocurrency.String())
+	chains, err := g.ListCurrencyChain(ctx, cryptocurrency.Upper())
 	if err != nil {
 		return nil, err
 	}
 
-	availableChains := make([]string, len(chains.MultichainAddresses))
-	for x := range chains.MultichainAddresses {
-		availableChains[x] = chains.MultichainAddresses[x].Chain
+	availableChains := make([]string, len(chains))
```

Comment:
With the disabled check below this creates zero values on the return, please use capacity in make and then append.